### PR TITLE
feat(edc):[TRI-1064] Add notification support for EDC

### PR DESCRIPTION
**Pull request by ds-jkreutzfeld**

### CODE REVIEW

File: edc-client/src/main/java/org/eclipse/tractusx/edc/EdcSubmodelClient.java [hunk 1]
```diff
@@ -52,6 +52,9 @@ CompletableFuture<List<Relationship>> getRelationships(String submodelEndpointAd
             RelationshipAspect traversalAspectType) throws EdcClientException;
 
     CompletableFuture<String> getSubmodelRawPayload(String submodelEndpointAddress) throws EdcClientException;
+
+    CompletableFuture<EdcNotificationResponse> sendNotification(final String submodelEndpointAddress,
+            final EdcNotification notification) throws EdcClientException;
```
Comment (ds-jhartmann) [inline edc-client/src/main/java/org/eclipse/tractusx/edc/EdcSubmodelClient.java:57]: Checkstyle error: redundant modifier 'final'

File: edc-client/src/main/java/org/eclipse/tractusx/edc/EdcSubmodelFacade.java [hunk 2]
```diff
@@ -76,4 +76,20 @@ public String getSubmodelRawPayload(final String submodelEndpointAddress) throws
...
+            return client.sendNotification(submodelEndpointAddress, notification).get();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            return null;
+        } catch (ExecutionException e) {
+            final Throwable cause = e.getCause();
+            if (cause instanceof EdcClientException exceptionCause) {
+                throw exceptionCause;
+            }
+            throw new EdcClientException(cause);
```
Comment (ds-jhartmann) [inline edc-client/src/main/java/org/eclipse/tractusx/edc/EdcSubmodelFacade.java:91]: PMD failure: Rule:PreserveStackTrace Priority:3 New exception is thrown in catch block, original stack trace may be lost.

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]
Checkstyle and PMD issues. See individual comments. Everything else looks good.

File: edc-client/src/main/java/org/eclipse/tractusx/edc/EdcSubmodelFacade.java (hunk 2 above)
Comment (ds-jkreutzfeld) [reply edc-client/src/main/java/org/eclipse/tractusx/edc/EdcSubmodelFacade.java:92]: Resolved

File: edc-client/src/main/java/org/eclipse/tractusx/edc/EdcSubmodelClient.java (hunk 1 above)
Comment (ds-jkreutzfeld) [reply edc-client/src/main/java/org/eclipse/tractusx/edc/EdcSubmodelClient.java]: Resolved

**Review by ds-jhartmann — APPROVED** [review]

**Pull request merged**